winpr/libwinpr/sspi/Kerberos/kerberos.c

```diff
@@ -1325,13 +1325,45 @@ static SECURITY_STATUS SEC_ENTRY kerberos_AcceptSecurityContext(
 		if (!kerberos_rd_tgt_token(&input_token, &target, NULL))
 			goto bad_token;
 
+		/*
+		 *  we're requested with target="TERMSRV/<host>@<REALM>" but we're gonna look
+		 *  at <host>$@<REALM> in the keytab (notice the $), so we build a new "target"
+		 *  string containing
+		 *
+		 *  sname      realm
+		 *  |          |
+		 *  v          v
+		 *  <host>$\x00<REALM>
+		 *
+		 */
 		if (target)
 		{
-			if (*target != 0 && *target != '@')
-				sname = target;
+			char* sname = strchr(target, '/');
```

Comment:
maybe easier to read as
```
if (target)
		{
			if (strncmp(target, "TERMSRV/", 8) != 0)
				goto cleanup;
			const char* sname = &target[8];
			const char* realm = strchr(sname, '@');
			if (!realm)
				goto cleanup;

			const intptr_t diff = realm - sname;
			realm++;

			char* str = NULL;
			size_t strlen = 0;
			(void)winpr_asprintf(&str, &strlen, "%.*s$@%s", diff, sname, realm);
			if (!str)
				goto cleanup;
			free(target);
			target = str;
		}
```

Reply:
Hum I'm gonna take inspiration of that, but the realm is not always present, so I must adapt that.

Reply:
@hardening just a suggestion though, your code is working as well.